scripts/sell_unused_slots.py

```diff
@@ -126,13 +134,26 @@ def _lambda_invocation_counts(self, function_names: Iterable[str]) -> Dict[str,
 
 
 def main(argv):
+    parser = argparse.ArgumentParser(description=__doc__,
+                                     formatter_class=AzulArgumentHelpFormatter,
+                                     add_help=True)
+    parser.add_argument('--dry-run',
+                        action='store_true',
+                        help='Report status without altering resources')
+    args = parser.parse_args(argv)
+
     # Listing BigQuery reservations is quicker than checking for an active
     # reindex, hence the order of checks
     slot_manager = SlotManager()
     if slot_manager.has_active_slots():
+        log.info('Slots are active.')
         monitor = ReindexDetector()
         if not monitor.is_reindex_active():
-            slot_manager.ensure_slots_deleted()
+            if args.dry_run:
+                log.info('Unused slots detected. Leaving them in place because '
+                         '--dry-run was passed.')
+            else:
+                slot_manager.ensure_slots_deleted()
```

Comment:
There is a conditional inside this method that would be nice to exercise in a dry run.

```suggestion
                slot_manager.ensure_slots_deleted(args.dry_run)
```

Same for `ensure_slots_active()` except that it may never be called with True.

Could also pass `args.dry_run` to the `SlotManager` constructor so it's available in all methods.